File: app/src/main/java/com/esafirm/androidplayground/dagger/example/ScopedInClass.java
package com.esafirm.androidplayground.dagger.example;


import com.esafirm.androidplayground.dagger.ActivityScope;

import javax.inject.Inject;

@ActivityScope
public class ScopedInClass {

    @Inject
    public ScopedInClass() {
    }
}

File: app/src/main/java/com/esafirm/androidplayground/dagger/example/ScopedInClass.kt
package com.esafirm.androidplayground.dagger.example


import com.esafirm.androidplayground.dagger.ControllerScope

import javax.inject.Inject
import kotlin.random.Random

/**
 * A sample of class that is scoped in class declaration
 */
@ControllerScope
class ScopedInClass @Inject constructor() {

    private val id = Random.nextInt()

    override fun toString(): String = "Scoped in class -- $id"
}
